Src/ILGPU/Frontend/ILFrontend.cs

```diff
@@ -160,20 +170,41 @@ private void DoWork()
                     "Invalid processing state");
 
                 detectedMethods.Clear();
-                codeGenerationPhase.GenerateCodeInternal(
-                    current.Method,
-                    current.IsExternalRequest,
-                    detectedMethods,
-                    out Method method);
-                current.SetResult(method);
+                try
+                {
+                    codeGenerationPhase.GenerateCodeInternal(
+                        current.Method,
+                        current.IsExternalRequest,
+                        detectedMethods,
+                        out Method method);
+                    current.SetResult(method);
+                }
+#pragma warning disable CA1031 // Do not catch general exception types
```

Comment:
Maybe we should convert the pragma to a `[Suppress...]` attribute?

Reply:
Done.